### PR TITLE
matrix-sliding-sync: drop

### PR DESCRIPTION
**Pull request by mweinelt**
On the 2024 matrix conference the EOL for the sliding-sync-proxy was announced to be 2024-10-15. While the repo does not yet reflect that state, we should not be taking the the sliding-sync proxy into NixOS 24.11 under any circumstances.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/matrix-synapse/sliding-sync/default.nix
Comment (emilylange) [inline pkgs/servers/matrix-synapse/sliding-sync/default.nix:1]: This needs to be removed from `all-packages.nix` too

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1026,6 +1026,7 @@ mapAliases {
   matrique = throw "'matrique' has been renamed to/replaced by 'spectral'"; # Converted to throw 2024-10-17
   matrixcli = throw "'matrixcli' has been removed due to being unmaintained and broken functionality. Recommend 'matrix-commander' as an alternative"; # Added 2024-03-09
   matrix-recorder = throw "matrix-recorder has been removed due to being unmaintained"; # Added 2023-05-21
+  matrix-sliding-sync = throw "matrix-sliding-sync has been removed as matrix-synapse 114.0 and later covers its functionality"; # Added 2024-10-20
```
Comment (emilylange) [inline pkgs/top-level/aliases.nix:1029]: Can you also add a release note entry for this?

**Review by emilylange — CHANGES_REQUESTED** [review]
Thanks! :)
This is something I simply didn't find time for yet.
I initially wanted to wait until F-Droid caught up with the newer Element-X versions that contain the migration hint, but that has since long happened soooo

I agree that we should drop it before branch-off.

Just the two review comments and there is a typo in your commit message: matri**d**-sliding-sync vs matri**x**-sliding-sync

**Comment by mweinelt** [comment]
Changes applied.

**Review by emilylange — APPROVED** [review]
lgtm, thanks :)

**Comment by Mic92** [comment]
For people that still want sliding sync with dendrite. Here is the revert: [aed993733203](https://github.com/NixOS/nixpkgs/commit/aed993733203b1b403e7ce8a4ab8255f9a1db456)

Dendrite doesn't receive many features these days, but it is still maintained until the funding situation improves: https://github.com/matrix-org/dendrite/issues/3413#issuecomment-2364564129

**Pull request merged**